public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -157,11 +160,55 @@ function AddAnomalyDetector({
 
   const notifications = getNotifications();
   const handleValidationAndSubmit = (formikProps) => {
-    if (!isEmpty(formikProps.errors)) {
-      focusOnFirstWrongFeature(formikProps.errors, formikProps.setFieldTouched);
-      notifications.toasts.addDanger('One or more input fields is invalid');
+    if (formikProps.values.featureList.length !== 0) {
+      formikProps.setFieldTouched('featureList', true);
+      formikProps.validateForm().then(async (errors) => {
+        if (!isEmpty(errors)) {
+          focusOnFirstWrongFeature(errors, formikProps.setFieldTouched);
+          notifications.toasts.addDanger(
+            'One or more input fields is invalid.'
+          );
+        } else {
+          const isAugmentationEnabled = uiSettings.get(
+            PLUGIN_AUGMENTATION_ENABLE_SETTING
+          );
+          if (!isAugmentationEnabled) {
+            notifications.toasts.addDanger(
+              'Visualization augmentation is disabled, please enable visualization:enablePluginAugmentation.'
+            );
+          } else {
+            const maxAssociatedCount = uiSettings.get(
+              PLUGIN_AUGMENTATION_MAX_OBJECTS_SETTING
+            );
+            await savedObjectLoader.findAll().then(async (resp) => {
```

Comment:
nit: why not use `getAugmentVisSavedObjs`?

Reply:
the return type of this function is not ideal, wanted to show warning toast message instead of that not well formatted error message